### PR TITLE
Pagination and searching activities (plus sync with original code and update Bosnian translation)

### PR DESCRIPTION
**Pull request by Kolesar**
Updated Bosnian language and added new functionalities on task activity:
- pagination
- searching with filters

Now task comment is searchable from searching task, or searching task activity.

### CODE REVIEW

**Comment by Kolesar** [comment]
Sorry for disturb you @fguillot, but I have a few more features ready on my machine. I do not want to push that on github, because I waiting for your approve current merge to your code. Please tell me do you want to merge this and when (I understand, you are probably really busy), if not, I will continue push code on github, but in that case we will have two different version of Kanboard on public/github.
If you are request me to make a some changes, please tell me, because I do not want to make anything parallel on github.
And again, sorry for disturb you and push you for answer ;)

**Comment by BlueTeck** [comment]
You should change the title of your pull request, it is not only the bosnian translation anymore.

**Comment by Kolesar** [comment]
@BlueTeck done ;)

**Comment by fguillot** [comment]
@Kolesar Thanks for submitting a pull-request. I'm ok to merge your PR but some improvements have to be made:

- Reduce the code duplication, especially in the class `ActivityFilter`
- Add at least some unit tests for the new filters

Thanks!

**Comment by Kolesar** [comment]
@fguillot I do not understand you about duplication code inside ``ActivityFilter``. All methods are very similar like in ``TaskFilter``, but have a difference and ``ActivityFilter`` should have own methods.

Can you be more specific please?

And, yes, I'll add unit test.

File: app/Model/ActivityFilter.php [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+    }
+
+    /**
+     * Get the event html content
+     *
+     * @access public
+     * @param  array     $params    Event properties
+     * @return string
+     */
+    public function getContent(array $params)
```
Comment (fguillot) [inline app/Model/ActivityFilter.php]: This method and the one below are duplicated from the class `ProjectActivity`, there is probably a better way of doing that.

**Comment by fguillot** [comment]
Thanks for updating your pull-request.

I was talking about the methods `getContent()` and `decode()`, it would be nice if you can avoid to duplicate that.

Thanks.

**Comment by Kolesar** [comment]
@fguillot Did you see solution about duplicate? I'm afraid that is not best solution. Can you make review code again, please? :)

**Comment by Kolesar** [comment]
And? :)

**Comment by fguillot** [comment]
I started to improve your PR in the branch "pr-review". I fixed some invalid code and a couple of bugs. It's not really ready to be merged in master.

I will will probably do a lot of code refactoring.

**Comment by Kolesar** [comment]
So, can you tell me what I should change? I really need this option (exactly my team need). I have ready many another features for Kanboard on my local machine and we use (all my team) these features in our Kanboard.

**Comment by fguillot** [comment]
As you can see in the branch master, I have done a lot of code refactoring to make the filters more flexible.

If you want to add more filters, you just need to follow the current mechanism.

Thanks.

**Pull request closed without merging**